src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -16,6 +21,63 @@ public void setup() {
 
     @Test
     public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+        assertFalse(productDao.findProducts().isEmpty());
+    }
+
+    @Test
+    public void testGetProduct(){
+        Optional<Product> product = productDao.getProduct(1L);
+        assertTrue(product.isPresent());
+    }
+
+    @Test
+    public void testDeleteProduct(){
+        Optional<Product> product = productDao.getProduct(2L);
+        assertTrue(product.isPresent());
+        productDao.delete(2L);
+        product = productDao.getProduct(2L);
+        assertFalse(product.isPresent());
+    }
+
+    @Test
+    public void testSaveProduct(){
+        Currency usd = Currency.getInstance("USD");
+        Product product = new Product("save test", "Samsung Galaxy SS", new BigDecimal(100), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
+        productDao.save(product);
+        assertTrue(productDao.getProduct(product.getId()).isPresent());
     }
```

Comment:
Please format the code here:
1)
public class ArrayListProductDaoTest
{

-->

public class ArrayListProductDaoTest {

2)
public void testDeleteProduct(){ -->
public void testDeleteProduct() {


Ctrl + Alt + L shortcut can be used in IntellIJ IDEA for automatic code formatting.